models/jira__issue_status_transitions.sql

```diff
@@ -16,21 +17,22 @@ issue_status_history as (
 
   select
     issue_id,
+    source_relation,
     status,
     status_category_name,
     valid_from,
     -- Recalculate valid_until based on next actual status change
     coalesce(
-      lead(valid_from) over (partition by issue_id order by valid_from),
+      lead(valid_from) over (partition by issue_id {{ jira.partition_by_source_relation() }} order by valid_from),
       {{ dbt.current_timestamp() }}
     ) as valid_until,
     previous_status,
 
     -- Current status indicator: if no next status change, this is current
-    case when lead(valid_from) over (partition by issue_id order by valid_from) is null then true else false end as is_current_status,
+    case when lead(valid_from) over (partition by issue_id {{ jira.partition_by_source_relation() }} order by valid_from) is null then true else false end as is_current_status,
 
     -- Calculate seconds in each status period using recalculated valid_until
-    cast({{ dbt.datediff('valid_from', 'coalesce(lead(valid_from) over (partition by issue_id order by valid_from), ' ~ dbt.current_timestamp() ~ ')', 'second') }} as {{ dbt.type_int() }}) as seconds_in_status
+    {{ dbt.datediff('valid_from', 'coalesce(lead(valid_from) over (partition by issue_id' ~ (', source_relation' if var('jira_sources', [])|length > 1 else '') ~ ' order by valid_from), ' ~ dbt.current_timestamp() ~ ')', 'second') }} as seconds_in_status
```

Comment:
We should use the `jira.partition_by_source_relation()`macro here. I think it'd look like

```suggestion
    {{ dbt.datediff('valid_from', 'coalesce(lead(valid_from) over (partition by issue_id ' ~ jira.partition_by_source_relation() ~ ' order by valid_from), ' ~ dbt.current_timestamp() ~ ')', 'second') }} as seconds_in_status
```

Reply:
updated